app/lib/analyzer/task_sources.dart

```diff
@@ -77,14 +77,17 @@ class DatastoreHistoryTaskSource implements TaskSource {
   Stream<Task> startStreaming() async* {
     for (;;) {
       try {
-        await for (PackageVersion pv in _db.query(PackageVersion).run()) {
-          await new Future.delayed(const Duration(seconds: 30));
+        final Query query = _db.query(PackageVersion)..order('-created');
+        await for (PackageVersion pv in query.run()) {
           final List<PackageVersionAnalysis> list = await _db.lookup([
             _db.emptyKey
                 .append(PackageAnalysis, id: pv.package)
                 .append(PackageVersionAnalysis, id: pv.version)
           ]);
           if (list.first == null) {
+            // Waiting to throttle the historical tasks. If there is a newly
+            // uploaded package, give it a chance before continuing with these.
+            await new Future.delayed(const Duration(seconds: 30));
```

Comment:
Since the sleep is orthongonal to other functionality, you could separate the concerns and make a wrapper:

```dart
class ThrottlingTaskSource implements TaskSource {
  final TaskSource innerSource;
  final Duration sleepDuration;

  ThrottlingTaskSource(this.innerTaskSource, this.sleepDuration);

  Stream<Task> startStreaming() async {
    await for (final Task task in innerSource.startStreaming()) {
      yield task;
      await new Future.delayed(sleepDuration);
    }
  }
}
```